utils/stall_free.hh

```diff
@@ -56,14 +58,99 @@ void merge_to_gently(std::list<T>& list1, const std::list<T>& list2, Compare com
     }
 }
 
-template<class T>
-seastar::future<> clear_gently(std::list<T>& list) {
-    return repeat([&list] () mutable {
-        if (list.empty()) {
-            return seastar::stop_iteration::yes;
-        }
-        list.pop_back();
-        return seastar::stop_iteration::no;
+template <typename T>
+concept GentlyClearable = requires (T x) {
+    { x.clear_gently() } -> std::same_as<future<>>;
+};
+
+template <typename Ptr, typename T = typename Ptr::element_type>
+concept SmartPointer = requires (Ptr x) {
+    { x.get() } -> std::same_as<T*>;
+};
+
+template <typename Vector, typename T = typename Vector::value_type>
+concept VectorClass = requires (Vector x) {
+    { x.empty() } -> std::same_as<bool>;
+    { x.back() } -> std::same_as<T&>;
+    { x.pop_back() } -> std::same_as<void>;
+};
+
+template <typename Container, typename Iterator = typename Container::iterator>
+concept ContainerClass = requires (Container x) {
+    { x.empty() } -> std::same_as<bool>;
+    { x.erase(x.begin()) } -> std::same_as<Iterator>;
```

Comment:
I think if you use the algorithm `x.erase(std::prev(x.end()));` then it works for both vector-like and associative containers.

Reply:
It works, but it requires the elements to be copyable since the algorithm is stupid about `erase` since it doesn't know it's the last element.  `pop_back` works much better in this case.
and  I still want to single out std::vector of trivially destructible elements since it can be cleared in bulk, so I'll stick to the vector-like concept (and rename it to `VectorLike` to make it clearer)

Reply:
Fixed in v5 (using `VectorLike` concept that covers also `small_vector` and `chunked_vector`)